src/get-started/java-binding.md

```diff
@@ -181,6 +194,12 @@ current storage state when the transaction is executed. Exonum passes `Fork`
 as an argument – a view that allows performing modifying operations. A service
 schema object can be used to access data collections of this service.
 
+Also, `Transaction#execute` method may throw `TransactionExecutionException` 
+which contains transaction error report. This feature allows the service to notify 
+Exonum about an error in a transaction execution whenever one occurs. 
+If transaction execution fails, the changes invoked by the transaction are rolled 
+back, while the error data is stored in the database for further user reference.
```

Comment:
Also, `Transaction#execute` method may throw `TransactionExecutionException`
which contains a transaction error report. This feature allows users to notify
Exonum about an error in a transaction execution whenever one occurs. The service checks the preconditions before executing a transaction according to its business logic. As a result, the service either accepts a transaction or, if the preconditions are not met, issues an exception that is further transformed into an Exonum core [TransactionResult enum]( https://docs.rs/exonum/0.9.4/exonum/blockchain/type.TransactionResult.html) containing a corresponding error code and a message with error data.

If transaction execution fails, the changes invoked by the transaction are rolled back, while the error data is stored in the database for further user reference. Light clients also provide access to information on the transaction error, stored in the Exonum storage, to their users.